File: coroutine-py/basics05.py
import asyncio


def main():
    asyncio.run(async_main())


async def async_main():
    async with asyncio.TaskGroup() as tg:
        task = tg.create_task(do_world())
        print('Hello')
        await task
        print('Done')


async def do_world():
    await asyncio.sleep(1)
    print('World!')


if __name__ == "__main__":
    main()

File: asyncawait-ts/basics05.ts
import {sleep} from "./lib";

function main() {
    (async () => {
        const p = (async () => {
            await sleep(1000);
            console.log('World!');
        })();
        console.log('Hello');
        await p;
        console.log('Done');
    })();
    // cannot block
}

main();
